packages/analytics-browser/src/config.ts

```diff
@@ -7,90 +7,95 @@ import {
+import { CookieStorage, getCookieName, FetchTransport, getQueryParams } from '@amplitude/analytics-client-common';
 
 import { LocalStorage } from './storage/local-storage';
 import { XHRTransport } from './transports/xhr';
 import { SendBeaconTransport } from './transports/send-beacon';
-
-export const getDefaultConfig = () => {
-  const cookieStorage = new MemoryStorage<UserSession>();
-  const trackingOptions: Required<TrackingOptions> = {
-    ipAddress: true,
-    language: true,
-    platform: true,
-  };
-  return {
-    cookieExpiration: 365,
-    cookieSameSite: 'Lax',
-    cookieSecure: false,
-    cookieStorage,
-    cookieUpgrade: true,
-    disableCookies: false,
-    domain: '',
-    sessionTimeout: 30 * 60 * 1000,
-    trackingOptions,
-    transportProvider: new FetchTransport(),
-  };
-};
-
+import { parseLegacyCookies } from './cookie-migration';
+import { CookieOptions } from '@amplitude/analytics-types/lib/esm/config/browser';
+import { DEFAULT_IDENTITY_STORAGE, DEFAULT_SERVER_ZONE } from './constants';
 export class BrowserConfig extends Config implements IBrowserConfig {
-  appVersion?: string;
-  cookieExpiration: number;
-  cookieSameSite: string;
-  cookieSecure: boolean;
-  cookieUpgrade: boolean;
-  cookieStorage: Storage<UserSession>;
-  defaultTracking?: DefaultTrackingOptions | boolean;
-  disableCookies: boolean;
-  domain: string;
-  partnerId?: string;
-  sessionTimeout: number;
-  trackingOptions: TrackingOptions;
-
-  // NOTE: These protected properties are used to cache values from async storage
+  protected _cookieStorage: Storage<UserSession>;
   protected _deviceId?: string;
   protected _lastEventId?: number;
   protected _lastEventTime?: number;
   protected _optOut = false;
   protected _sessionId?: number;
   protected _userId?: string;
 
-  constructor(apiKey: string, options?: BrowserOptions) {
-    const defaultConfig = getDefaultConfig();
-    super({
-      flushIntervalMillis: 1000,
-      flushMaxRetries: 5,
-      flushQueueSize: 30,
-      transportProvider: defaultConfig.transportProvider,
-      ...options,
-      apiKey,
-    });
+  constructor(
+    public apiKey: string,
+    public appVersion?: string,
+    cookieStorage: Storage<UserSession> = new MemoryStorage(),
+    public cookieOptions: CookieOptions = {
+      domain: '',
+      expiration: 365,
+      sameSite: 'Lax' as const,
+      secure: false,
+      upgrade: true,
+    },
+    public defaultTracking: boolean | DefaultTrackingOptions = true,
+    deviceId?: string,
+    public flushIntervalMillis: number = 1000,
+    public flushMaxRetries: number = 5,
+    public flushQueueSize: number = 30,
+    public identityStorage: IdentityStorageType = DEFAULT_IDENTITY_STORAGE,
+    public ingestionMetadata?: IngestionMetadata,
+    lastEventId?: number,
+    lastEventTime?: number,
+    public previousSessionDeviceId?: string,
+    public previousSessionUserId?: string,
```

Comment:
So we have previousSessionUserId and previousSessionDeviceId in config now? and we also have userId I believe, what's the major difference between these two?

Reply:
I think this is used to assign the right sessionId and deviceId for the session end event. Check [here](https://github.com/amplitude/Amplitude-TypeScript/blob/browser-sdk-v2/packages/analytics-browser/src/browser-client.ts#L190-L206).
Also curious why we need to include those two in the config.

Reply:
Hey, I moved it to config to consolidate it with the other metadata like `lastEventId`, `lastEventTime` which I believe serve similar purposes. What I made sure is that these are not included in public facing options on `init()`

Internal config definition: https://github.com/amplitude/Amplitude-TypeScript/blob/d1f528862211b96b13e4e3539338de76914699b6/packages/analytics-types/src/config/browser.ts#L23

 Browser options extends external config only: https://github.com/amplitude/Amplitude-TypeScript/blob/d1f528862211b96b13e4e3539338de76914699b6/packages/analytics-types/src/config/browser.ts#L63

Reply:
That makes sense to me. Thanks.